Makefile

```diff
@@ -1,8 +1,18 @@
 CC=g++
 CFLAGS=-Wall -g
+DEPS=pso.h
 
-default: all
-all: clean demo
-demo: demo.c pso.c
+SOURCE_FILES=$(shell find -name '*.c')
```

Comment:
Tested this locally and I have one more small fix: `find . -name '*.c'` instead of `find -name '*.c'`. 

Reason is compatibility with BSD find (for macos builds) in addition to GNU find.

Thanks again!

Reply:
okay nice .. i didn't know